tests/integration/istioctl/istioctl_test.go

```diff
@@ -0,0 +1,93 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package istioctl
+
+import (
+	"bytes"
+	"regexp"
+	"strings"
+	"testing"
+
+	"istio.io/istio/istioctl/cmd"
+
+	"istio.io/istio/pkg/test/framework"
+	"istio.io/istio/pkg/test/framework/components/environment"
+	"istio.io/istio/pkg/test/framework/components/environment/kube"
+	"istio.io/istio/pkg/test/framework/components/galley"
+	"istio.io/istio/pkg/test/framework/components/istio"
+	"istio.io/istio/pkg/test/framework/components/pilot"
+	"istio.io/istio/pkg/test/framework/label"
+	"istio.io/istio/pkg/test/framework/resource"
+)
+
+var (
+	i   istio.Instance
+	env *kube.Environment
+)
+
+// This test requires `--istio.test.env=kube` because it tests istioctl doing PodExec
+func TestMain(m *testing.M) {
+	framework.
+		NewSuite("istioctl_integration_test", m).
+		RequireEnvironment(environment.Kube).
+		Label(label.Presubmit).
+
+		// Deploy Istio
+		SetupOnEnv(environment.Kube, istio.Setup(&i, nil)).
+		SetupOnEnv(environment.Kube, func(ctx resource.Context) error {
+			env = ctx.Environment().(*kube.Environment)
+			return nil
+		}).
+		Run()
+}
+
+// TestVersion does "istioctl version --remote=true" to verify the CLI understands the data plane version data
+func TestVersion(t *testing.T) {
+	framework.
+		NewTest(t).
+		Run(func(ctx framework.TestContext) {
+			g := galley.NewOrFail(t, ctx, galley.Config{})
+			_ = pilot.NewOrFail(t, ctx, pilot.Config{Galley: g})
+
+			args := []string{"version", "--remote=true"}
+
+			var out bytes.Buffer
+			rootCmd := cmd.GetRootCmd(args)
+			rootCmd.SetOutput(&out)
+			fErr := rootCmd.Execute()
```

Comment:
Can I recommend adding a test framework component for istioctl itself? This allows others to use istioctl as part of their tests, if need be.
https://github.com/istio/istio/tree/master/pkg/test/framework/components contains the current components.
This is a mock/example component: https://github.com/istio/istio/tree/master/tests/integration/framework/mycomponent